### PR TITLE
ON-8480 # Added new files element with file upload to s3 

### PR DESCRIPTION
**Pull request by Zaxist**

### CODE REVIEW

File: src/form-elements/FormElementFiles/FormElementFiles.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+      const newAttachments = await addAttachments(Array.from(newFiles))
+      const updatedValues = Array.isArray(value) ? [...value] : []
+      updatedValues.push(...newAttachments)
+      if (isMounted.current) {
+        onChange(element, updatedValues)
+        setIsDirty()
+      }
+    },
+    [form, addAttachments, value, isMounted, onChange, element, setIsDirty],
```
Comment (mymattcarroll) [inline src/form-elements/FormElementFiles/FormElementFiles.tsx:53]: Why is `form` here?

File: src/form-elements/FormElementFiles/index.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    formElement: FormTypes.FormElement,
+    newValue: PossibleFileConfiguration[],
+  ) => unknown
+  displayValidationMessage: boolean
+  validationMessage: string | undefined
+}
+
+const FormElementFilesController = ({ value, onChange, ...props }: Props) => {
+  if (!props.element.storageType || props.element.storageType === 'legacy') {
+    console.log('Using legacy files...')
```
Comment (mymattcarroll) [inline src/form-elements/FormElementFiles/index.tsx:23]: Logs in render in not recommended.

File: src/form-elements/FormElementFiles/index.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          onChange as (
+            formElement: FormTypes.FormElement,
+            newValue: FilesElementFile[],
+          ) => unknown
+        }
+        {...props}
+      />
+    )
+  }
+  console.log('Using new files...')
```
Comment (mymattcarroll) [inline src/form-elements/FormElementFiles/index.tsx:37]: Same same.

File: src/form-elements/FormElementFiles/FormElementFiles.tsx (hunk 1 above)
Comment (Zaxist) [reply src/form-elements/FormElementFiles/FormElementFiles.tsx:53]: Shouldn't be anymore. I'll remove

File: src/form-elements/FormElementFiles/index.tsx (hunk 2 above)
Comment (Zaxist) [reply src/form-elements/FormElementFiles/index.tsx:23]: Oops

File: src/hooks/attachments/useAttachment.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        const upload = await submissionService.uploadAttachment({
+          formId: form.id,
+          file: {
+            name: newAttachment.fileName,
+            type: newAttachment.data.type,
+            data: newAttachment.data,
+            isPrivate,
+          },
+        })
+        throw new Error('An error occured')
```
Comment (mymattcarroll) [inline src/hooks/attachments/useAttachment.ts:47]: Just need to remove this.

Comment (Zaxist) [reply src/hooks/attachments/useAttachment.ts:47]: Oops

**Pull request merged**